bcipy/helpers/visualization.py

```diff
@@ -1,73 +1,77 @@
+from os import path
+from typing import List, Optional
+
 from matplotlib.figure import Figure
+from mne.io import read_raw_edf
 import matplotlib.pyplot as plt
+import numpy as np
+
 from bcipy.helpers.load import choose_csv_file, load_raw_data
-from mne.io import read_raw_edf
-from typing import List
 
-import logging
 log = logging.getLogger(__name__)
 
 
-def generate_offline_analysis_screen(
-        x,
-        y,
-        model=None,
-        folder=None,
-        save_figure=True,
-        down_sample_rate=2,
-        fs=300,
-        plot_x_ticks=8,
-        plot_average=False,
-        show_figure=False,
-        channel_names=None) -> List[Figure]:
-    """ Offline Analysis Screen.
+def visualize_erp(
+        data,
+        labels,
+        fs,
+        class_labels: List[str] = ['Non-Target', 'Target'],
+        plot_average: bool = False,
+        show_figure: bool = False,
+        save_path: Optional[str] = None,
+        figure_name: str = 'mean_erp.pdf',
+        channel_names: Optional[dict] = None) -> Figure:
+    """ Visualize ERP.
 
-    Generates the information figure following the offlineAnalysis.
-    The figure has multiple tabs containing the average ERP plots.
+    Generates a comparative ERP figure following a task execution. Given a set of trailed data,
+    and labels describing two classes, they are plotted and may be saved or shown in a window.
 
     Returns a list of the figure handles created.
 
     PARAMETERS
     ----------
-
-    x(ndarray[float]): C x N x k data array
-    y(ndarray[int]): N x k observation (class) array
-        N is number of samples k is dimensionality of features
-        C is number of channels
-    model(): trained model for data
-    folder(str): Folder of the data
-    save_figures: boolean: whether or not to save the plots as PDF
-    down_sample_rate: downsampling rate applied to signal (if any)
-    fs (sampling_rate): original sampling rate of the signal
-    plot_x_ticks: number of ticks desired on the ERP plot
+    N: samples
+    k: features
+    C: channels
+
+    data(ndarray[float]): C x N x k data array
+    labels(ndarray[int]): N x k observation (class) array. Assumed to be two classes [0, 1]
+    fs (sampling_rate): sampling rate of the current data signal
+    class_labels: list of legend names for the respective classes for plotting (0 ,1).
     plot_average: boolean: whether or not to average over all channels
     show_figure: boolean: whether or not to show the figures generated
+    save_path: optional path to a save location of the figure generated
+    figure_name: name of the figure to be used when save_path provided
     channel_names: dict of channel names keyed by their position.
     """
-    fig_handles = []
-
     channel_names = channel_names or {}
-    classes = np.unique(y)
+    classes = np.unique(labels)
 
-    means = [np.squeeze(np.mean(x[:, np.where(y == i), :], 2))
+    means = [np.squeeze(np.mean(data[:, np.where(labels == i), :], 2))
```

Comment:
It's a lot more readable using `data` and `labels` as the variable names. While we're at it, maybe rename `i` to something more descriptive? `i` usually indicates an index, but here it's the class label.